File: fixtures/kotlin/complexity.kt
fun simple() {
    println("hi")
}

fun branchy(x: Int): Int {
    if (x > 0) {
        return 1
    } else if (x < -10) {
        return -2
    }
    for (i in 0 until x) {
        while (i > 2) {
            break
        }
    }
    var y = x
    do {
        y--
    } while (y > 5)
    when (x) {
        1 -> return 1
        2 -> return 2
        else -> {}
    }
    val label: String? = null
    val fallback = label ?: "none"
    if (y > 1 && y < 100 || x == -5) {
        return y
    }
    return try {
        canThrow()
        fallback.length
    } catch (error: Exception) {
        -1
    }
}

fun canThrow() {}

class Shape(var width: Int) {
    var count: Int = 0

    val area: Int
        get() {
            return if (width > 0) width * width else 0
        }

    var label: String = ""
        get() = if (field.isEmpty()) "none" else field
        set(value) {
            if (value.isNotEmpty()) {
                field = value
            }
        }

    init {
        if (width < 0) {
            width = 0
        }
    }

    constructor(width: Int, count: Int) : this(width) {
        if (count > 0) {
            this.count = count
        }
    }

    fun describe(): String {
        return if (area > 10) "big" else "small"
    }

    companion object {
        fun unit(): Shape {
            return Shape(1)
        }
    }
}

object Registry {
    fun register(shape: Shape): Boolean {
        return shape.area > 0
    }
}

interface Labeled {
    fun label(size: Int): String {
        return if (size > 10) "big" else "small"
    }
}

File: fixtures/javascript/complexity.js
function simple() {
  console.log("hi");
}

function branchy(x) {
  if (x > 0) {
    return 1;
  } else if (x < -10) {
    return -2;
  }
  for (let i = 0; i < x; i++) {
    while (i > 2) {
      break;
    }
  }
  let y = x;
  do {
    y--;
  } while (y > 5);
  for (const n of [1, 2, 3]) {
    if (n === y) {
      break;
    }
  }
  switch (x) {
    case 1:
      return 1;
    case 2:
      return 2;
    default:
      break;
  }
  const label = null;
  const fallback = label ?? "none";
  if ((y > 1 && y < 100) || x === -5) {
    return y;
  }
  try {
    canThrow();
    return fallback.length;
  } catch (error) {
    return -1;
  }
}

function canThrow() {}

const double = (n) => (n > 0 ? n * 2 : 0);

class Shape {
  constructor(width) {
    this.width = width;
  }

  get area() {
    return this.width > 0 ? this.width * this.width : 0;
  }

  set area(value) {
    if (value > 0) {
      this.width = value;
    }
  }

  describe() {
    return this.area > 10 ? "big" : "small";
  }
}
